scripts/clean-shrinkwrap.js

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable no-undefined */
+
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const _ = require('lodash')
+
+const JSON_INDENT = 2
+const SHRINKWRAP_FILENAME = path.join(__dirname, '..', 'npm-shrinkwrap.json')
+
+const packageInfo = require('../package.json')
+const shrinkwrap = require('../npm-shrinkwrap.json')
+
+/**
+ * @summary Delete a dependency
+ *
+ * @param {*} keys - Keys reference
+ * @param {*} tree - Tree to iterate over
+ * @param {*} onNode - Function to apply
+ * @param {*} parentName - Name of dependent
+ * @param {*} depType - Dependency type (dependencies|requires)
+ *
+ * @example
+ * removeDependency(keys, tree, onNode, parentName)
+ */
+const removeDependency = (keys, tree, onNode, parentName, depType) => {
+  let name = null
+  for (let index = 0; index < keys.length; index += 1) {
+    name = keys[index]
+
+    // Check for this dependency to still exist,
+    // as a node might have been removed just before this iteration
+    if (tree[depType][name]) {
+      onNode(tree, parentName || tree.name, name, tree[depType][name], depType)
+
+      // Check that the walking function didn't remove the dependency
+      // if so, skip it and continue with the next one
+      if (tree[depType][name] && tree[depType][name][depType]) {
+        traverseDeps(tree[depType][name], onNode, name)
+      }
+    }
+  }
+}
+
+/**
+ * @summary Traverse a shrinkwrap tree and call
+ * a given function on each dependency
+ * @param {Object} tree - shrinkwrap
+ * @param {Function} onNode - callback({Object} parent, {String} parentName, {String} name, {Object} info)
+ * @param {String} [parentName] - name of dependent (used internally)
+ * @example
+ * traverseDeps(shrinkwrap, (parent, parentName, name, info) => {
+ *   // ...
+ * })
+ */
+const traverseDeps = (tree, onNode, parentName) => {
+  if (!tree.dependencies && !tree.requires) {
+    return
+  }
+
+  const depsKeys = Object.keys(tree.dependencies)
+  removeDependency(depsKeys, tree, onNode, parentName, 'dependencies')
+
+  if (_.isObject(tree.requires)) {
+    const reqsKeys = Object.keys(tree.requires)
+    removeDependency(reqsKeys, tree, onNode, parentName, 'requires')
+  }
+}
+
+console.log('Cleaning shrinkwrap...')
+
+// Walk the generated shrinkwrap tree & apply modifications if necessary
+traverseDeps(shrinkwrap, (parent, parentName, name, info, depType) => {
+  // If this dependency depends on a "blacklisted" optional
+  // dependency; remove it from the shrinkwrap
+  if (packageInfo.platformSpecificDependencies.includes(name)) {
+    console.log(' - Removing "%s" from "%s"', name, parentName)
+    parent.dependencies[name] = undefined
+    Reflect.deleteProperty(parent.dependencies, name)
+
+    // Check that we're not in the root object
+    // before removing the "requires" property
+    if (_.isObject(parent.requires)) {
+      parent.requires[name] = undefined
+      Reflect.deleteProperty(parent.requires, name)
```

Comment:
just use `delete parent.requires[name]`

Reply:
@zvin Linter complains though, suggesting to use `Reflect`